fix: Docker-Rehaul

spark-c opened the pull request:
This PR moderately restructures and adjusts the content for setting up a self-hosted instance of Appsmith using Docker.

1. The most prominent adjustment to the organization of the content is creating a **clearer divide** between the instructions for setting up **with docker-compose** vs **without docker-compose**. Each type of setup lives under its own 2nd level heading, and the sections are also set apart by an <hr/ . Accordingly, the _sub_-headings for these sections no longer specify "(with/without docker-compose)", as it was previously a bit messy.

2. In the docker-compose setup instructions, there were a couple of separate small **Business Edition** "info"s that were combined and moved slightly to create a single callout for helping EE users at a place in the steps where they'd need to do a couple of extra actions. There was also a link for "upgrading to business edition" docs later on this page; I merged it into this blurb as well since it is early in the content and likely that someone will find it while looking for those linked docs.

3. There were a handful of smaller errors in the technical details that have been fixed

4. Other misc shuffling, grammar, and clarity

Resolves appsmithorg/appsmith#16298

spark-c commented:
Another small restructure in progress for business vs community versions

spark-c commented:
Per review comments, organized **without docker-compose** section into Community Edition and Business Edition parts. Also added file icon for the docker-compose.yml file download, and made small note that docker-compose is recommended.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
````diff
@@ -5,172 +5,157 @@ sidebar_position: 1
 ```
 
-## Explore Appsmith (without docker-compose)
+:::info
````
This piece of info seems misplaced

spark-c replied:
This info about auto-backups is placed here because they are directly related to the auto-updates feature; before any auto-update, Appsmith makes a backup. (And scheduled backups aren't implemented yet, auto-backup only works as a part of auto-update)

Users may be hesitant to enable auto-updates in case their apps will get broken by an update that they didn't have control over. But if they see this info and read about the auto-backups, they can be sure that there is a failsafe in the case of breaking changes.

Does that make sense? I like the `:::info` there, but I'm open to other ideas if it should be somewhere else.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
```diff
@@ -179,16 +164,39 @@ docker rm -f appsmith
-* [Configuring Self Hosted Instances](../../instance-configuration/#configuring-docker-installations)
+* [Configuring Self-Hosted Instances](../../instance-configuration/#configuring-docker-installations)
 * [Managing the Appsmith instance](../../instance-management/)
 * [Tutorials](../../../../learning-and-resources/tutorials/)
```
Let's remove the Tutorials link from here

danciaclara replied:
I think Tutorials still shows up there

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
```diff
@@ -5,172 +5,157 @@ sidebar_position: 1
-
-:::note
-You can expect one container running when using the `docker-compose ps` command. Something like:
+## Setup with docker-compose
```
Isn't it just Docker Compose and not docker-compose?

spark-c replied:
Good point, I've been using `docker-compose` successfully but according to [this StackOverflow question](https://stackoverflow.com/questions/66514436/difference-between-docker-compose-and-docker-compose) it looks like the original `docker-compose` plugin is being deprecated and its features are being integrated into the newer Docker Desktop, which uses the `docker compose` syntax.

@sum35h or @sharat87 , do you know if it's safe to update all references to `docker-compose` and replace them with `docker compose`? Just want to make sure the behavior should be the same since I'm not super experienced with Docker.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
````diff
@@ -5,172 +5,157 @@ sidebar_position: 1
 ```
 
-You should see a message `Appsmith is Running!` once the container is ready. The message is also logged and available in server logs(<mark>`stacks/logs/backend/backend.log`</mark>).
+#### Updating
````
Here I think we should write "Updating Appsmith"

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
````diff
@@ -5,172 +5,157 @@ sidebar_position: 1
 ```
 
-#### Business Edition
+#### Updating
````
Here I think we should write "Updating Appsmith"

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
```diff
@@ -5,172 +5,157 @@ sidebar_position: 1
-:::
-
-### Updating Appsmith (with docker-compose)
+### Updating Appsmith manually
```
I think we can skip the word "manually" here as it is implied

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
```diff
@@ -5,172 +5,157 @@ sidebar_position: 1
-:::
-
-If this is your first time using docker, you should expect a welcome page similar to the one below.
+Congratulations, your Appsmith server should be up and running now. You can access it at [http://localhost](http://localhost). If this is your first time setting up using Docker, you should expect a welcome page similar to the one below.
```
The congratulations here seems unnecessary. IMO 'The Appsmith instance should be up and running now' should suffice.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
````diff
@@ -5,172 +5,157 @@ sidebar_position: 1
-```
-docker run -d --name appsmith -p 80:80 -v "$PWD/stacks:/appsmith-stacks" -e APPSMITH_LICENSE_KEY=<Add_YOUR_LICENSE_KEY> --pull always appsmith/appsmith-ee
-```
+### Community Edition setup
````
remove setup from here as it is already mentioned in the main heading. Do the same for Business.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
```diff
@@ -5,172 +5,157 @@ sidebar_position: 1
-    </a>
-</div>
-copy the script to your installation folder and make it executable
+Once the download is complete, the server should be up in under a minute. You can follow the logs with the following command:
```
This same piece of information in repeated thrice in the document. I suggest we create a new section just above/below 'Restarting Containers' section named  'Docker Logs' and have this information about both logs and backend logs info mentioned under here

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
```diff
@@ -179,16 +164,38 @@ docker rm -f appsmith
 ## Troubleshooting
 
-If you encounter any errors during this process, check out our guide on [debugging deployment errors](../../../../help-and-support/troubleshooting-guide/deployment-errors.md). If you are still facing any issues, please reach out to [support@appsmith.com](mailto:support@appsmith.com) or join our [Discord Server](https://discord.com/invite/rBTTVJp) to speak to the Appsmith team directly!
+If you encounter any errors during this process, try following this guide on [debugging deployment errors](../../../../help-and-support/troubleshooting-guide/deployment-errors.md). If you are still facing issues, please contact [support@appsmith.com](mailto:support@appsmith.com) or join the Appsmith [Discord Server](https://discord.com/invite/rBTTVJp) to speak to the Appsmith team directly.
```
replace 'try following this' with 'follow the guide on...'

sharat87 reviewed (commented):
Looks fairly good. Made some minor recommendations.

There's also a conflict with the `main` branch. You'll need to merge the `main` branch into your branch before you can merge this PR.

spark-c commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
```diff
@@ -5,172 +5,157 @@ sidebar_position: 1
-:::info
-Currently, auto-update is disabled on the docker-compose file. If you want to enable auto-update for Appsmith, please uncomment all the commented lines in the docker-compose file.
-:::
+Click the link below to download the `docker-compose.yml` file, and place it into your Appsmith installation folder.
```
I'm fine with either way -- Tagging @jnikhila since I remember speaking with her about the file icon vs hyperlink also.

Should we make a ticket or something to work on creating/finding a more prominent icon? We don't have a lot of file downloads, but I do agree that the icon doesn't stand out and is easy to miss.

I'll use link text for now.

jnikhila replied:
@spark-c / @danciaclara - there is already a CSS class that we are using for this file download icon. We can modify that to make it more prominent. I still feel putting a link is confusing as it looks like an anchor, and having a file download icon will be more helpful.

danciaclara replied:
I feel we should go with a link instead of icon because it could be missed if it is separate from the text. especially the case that it is today (users don't know that they have to click that icon to first download the script)
<img width="841" alt="Screenshot 2022-11-23 at 18 43 07" src="https://user-images.githubusercontent.com/32227135/203555493-ed66d7fb-cbde-4836-acb4-34a036919bac.png">

spark-c replied:
@jnikhila @danciaclara  I think I'm also leaning towards the text, particularly because the link text is:

> Click to [download the `docker-compose.yml` file](/img/docker-compose_(3).yml)

Since there is the "Click" instruction along with the highlighted "download \<scriptname\>", I think the intentions should be reasonably clear to the user.

I don't feel super strongly either way and I'm more than happy to go with whichever method; maybe we could loop in a web UX person for their opinion? I feel bad that this PR is already so old though, I don't want to hold it up too much.

jnikhila replied:
> I'm fine with either way -- Tagging @jnikhila since I remember speaking with her about the file icon vs hyperlink also.
> 
> 
> 
> Should we make a ticket or something to work on creating/finding a more prominent icon? We don't have a lot of file downloads, but I do agree that the icon doesn't stand out and is easy to miss.
> 
> 
> 
> I'll use link text for now.

Yes, let's go ahead with text. We can revisit if it's not obvious for users and if issue is raised on the doc.

spark-c commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
```diff
@@ -5,172 +5,157 @@ sidebar_position: 1
-For the **Business Edition** - change the **image** name from `appsmith-ce` to `appsmith-ee` for the `image:` key in the `docker-compose.yml` file.
-:::
+:::info
+**For Business Edition users:**
```
It is important for business users -- it exists within an info callout b/c the `:::info` stands out well on the page, and it can be simply inserted at the point in the steps where a business user needs to take one extra step in setup.

I can remove the styling if you like, just let me know -- I updated the bold text to `Business Edition`

spark-c commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
````diff
@@ -5,172 +5,157 @@ sidebar_position: 1
 ```
 
-If it is not available locally, the command above will download the Docker images and start the services. You can follow the logs with the following command:
+If the image doesn't exist locally, this command downloads the Docker images and starts its services. You can follow the logs with the following command:
````
Attempted a bit of a compromise on this one, because our new grammar linter disallows using `(s)`, and the alternative would be quite wordy I think. Let me know if this detail is a sticking point.

> If the image doesn't exist locally, this command downloads the necessary Docker image and starts the container.

spark-c commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
````diff
@@ -5,172 +5,157 @@ sidebar_position: 1
 ```
 
-You should see a message `Appsmith is Running!` once the container is ready. The message is also logged and available in server logs(<mark>`stacks/logs/backend/backend.log`</mark>).
+Once the container is ready, you should see `Appsmith is Running!` in the Docker logs. This message is also logged and available in server logs in (<mark>`stacks/logs/backend/backend.log`</mark>).
````
Ah, this must be an artifact from Gitbook

sharat87 reviewed (dismissed):
Approving, but still highlighting a couple minor things.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.md:
```diff
@@ -5,172 +5,157 @@ sidebar_position: 1
 
 * [Docker](https://docs.docker.com/get-docker/) (version 20.10.7 or later)
-* [Docker-Compose ](https://docs.docker.com/compose/install/)(version 1.29.2 or later)
+* [Docker-Compose](https://docs.docker.com/compose/install/) (version 1.29.2 or later) _(Recommended)_
```
I agree with @sharat87 I think there was confusion here. @spark-c Let's remove the recommended tag from the Docker compose under prerequisites. But I think we need to let users know that we recommend Docker compose commands as opposed to Docker run,@sharat87 could you confirm this? If that is the case then we can specify recommended tag for the Docker-compose configuration heading.

spark-c commented:
@sharat87 @danciaclara Hi, sorry for the confusion on the conversations being marked resolved! I'm not sure what happened to have resolved them, I may have done that by mistake because I don't remember having seen it.

I like the decisions we have settled on very much, I will make the changes now.
- Making `docker-compose` _Optional_ instead of _Recommended_
- Moving `docker-compose` below Docker since it is optional and requires Docker in the first place.
- I'll also go ahead and preemptively add _(recommended)_ to the "Setup with docker compose" heading, since that set of instructions is much simpler and easier for users to follow. I can revert it if needed.

jnikhila commented:
@spark-c - This looks amazing. Just a minor comment. Let's have the following structure:

Docker
├── Introduction
		└── Setup Docker
				└── Tabs
			      			└── Community Edition
                                            			└── Installation
                                            			└── Update
                             			└──  Business Edition
                                            			└── Installation
                                     			       └── Update
		└── Setup Docker Compose
				└── Tabs
			      			└── Community Edition
                                            			└── Installation
                                            			└── Update
                             			└──  Business Edition
                                            			└── Installation
                                     			       └── Update
		└── Restart Containers
		└── Troubleshooting
		└── Next Steps
The idea is that Community and Business Edition tabs can have all the heads and subheads related to a subtopic in the doc. 
Happy to discuss.

jnikhila reviewed (changes requested):
Let's restructure the doc a bit:
├── Introduction
		└── Setup Docker
				└── Tabs
			      			└── Community Edition
                                            			└── Installation
                                            			└── Update
                             			└──  Business Edition
                                            			└── Installation
                                     			       └── Update
		└── Setup Docker Compose
				└── Tabs
			      			└── Community Edition
                                            			└── Installation
                                            			└── Update
                             			└──  Business Edition
                                            			└── Installation
                                     			       └── Update
		└── Restart Containers
		└── Troubleshooting
		└── Next Steps

jnikhila commented:
-- The structure at the moment is correct. Let's keep it as is.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+
+Your Appsmith server should be up and running soon. You can access it at [http://localhost](http://localhost). You can also verify that it's working correctly in the [Docker logs](#docker-logs).
+
+If this is your first time setting up using Docker, you should expect a welcome page similar to the one below.
```
I think this welcome page info is unnecessary, especially with the screenshot and all (this could change in the future). If we are including this, let's remove the pronouns 

> If this is the first time that the instance has been set up using Docker, a welcome page will be displayed.


@jnikhila @sharat87 what are your thoughts?

spark-c replied:
quoted this message below

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+
+### Updating Appsmith
+
+To update Appsmith (configured with docker-compose) manually, go to the root directory of your installation and run the following command:
```
is 'configured with docker-compose' necessary, we have this under that section itself right?
Also replace 'your' with 'the'

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+
+Automatic updates are turned off by default in the docker-compose.yml file. To enable them, please follow these steps:
+
+1. Go to the root directory of your Appsmith instance and run:
```
replace 'your' with 'the'. and maybe because you are mentioning directory, replace 'instance' with 'installation'

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
````diff
@@ -0,0 +1,199 @@
+   docker-compose down
+   ```
+
+2. Open the `docker-compose.yml` file with any text editor and un-comment all the lines that are commented out (lines 13-23).
````
2. Open the `docker-compose.yml` file with any text editor and uncomment lines 13-23.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+To complement automatic updates, Appsmith also has an auto-backup feature. Before an automatic update, Appsmith first creates a backup to ensure you can [roll back](/getting-started/setup/instance-management/appsmithctl#restore-appsmith-instance) an update to a previous version of Appsmith if necessary. For this auto-backup to work, please ensure that there is a minimum of 2 GB of free storage; o ...
+:::
+
+This configuration runs an Appsmith instance and a Watchtower instance to keep Appsmith automatically up-to-date. Your installation is now configured to stay up-to-date automatically.
```
replace 'Your' with 'The'

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+   <TabItem label="Business Edition" value="business">
+
+:::note
+To upgrade your existing Community Edition installation to the Business Edition, [follow these instructions](/getting-started/setup/upgrade-to-business-edition/docker).
```
To upgrade Community Edition to Business Edition, [follow these instructions](/getting-started/setup/upgrade-to-business-edition/docker).

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+To upgrade your existing Community Edition installation to the Business Edition, [follow these instructions](/getting-started/setup/upgrade-to-business-edition/docker).
+:::
+
+1. Click to [download the `docker-compose.yml` file](/img/docker-compose_(3).yml), and place it into your Appsmith installation folder.
```
Let's avoid hyperlinking a lot of text

I think the above should just be

> Download the [docker-compose.yml](https://github.com/img/docker-compose_(3).yml) file, and place it in the Appsmith installation folder.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+
+1. Click to [download the `docker-compose.yml` file](/img/docker-compose_(3).yml), and place it into your Appsmith installation folder.
+
+**OR,** run the following cURL:
```
same numbering alignment comment as in the Community edition

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+
+2. Change the **image** name from `appsmith-ce` to `appsmith-ee` for the `image:` key in the `docker-compose.yml` file.
+
+3. Add your **license key** (`APPSMITH_LICENSE_KEY`) to the `environment:` key in the `docker-compose.yml` file.
```
replace 'your' with 'the'

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+
+3. Add your **license key** (`APPSMITH_LICENSE_KEY`) to the `environment:` key in the `docker-compose.yml` file.
+
+4. Start the Docker container by running the command below. You may need to run with `sudo` if `docker` and `docker-compose` aren't accessible by your user. If the image doesn't exist locally, this command downloads the necessary Docker image and starts the container.
```
Avoid 'you' and 'your'

Start the Docker container by running the command below. This may need to be run with `sudo` if `docker` and `docker-compose` aren't accessible by the user. If the image doesn't exist locally, this command downloads the necessary Docker image and starts the container.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+
+## Setup with Docker run
+
+To get Appsmith up and running:
```
remove this line

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+<Tabs groupId="editions">
+   <TabItem label="Community Edition" value="community">
+
+To update Appsmith manually, go to the root directory of your installation and run the following commands:
```
replace 'your' with 'the'

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+   </TabItem>
+</Tabs>
+
+Once the download is complete, the server should be up in under a minute. You can verify that it's working correctly in the [Docker logs](#docker-logs).
```
Let's avoid 'under a minute' if we don't know for sure

> Once the download is complete, the server should be up and running. Check [Docker logs] to verify that the instance is working correctly.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+
+## Docker logs
+
+Once your Appsmith container is ready, you should see `Appsmith is Running!` in the Docker logs.
```
Rewrite as 

> Once the Appsmith container is ready, the message `Appsmith is Running!` will be displayed in the Docker logs.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+
+## Restarting containers
+
+If your containers are failing to restart, you can [download and run the `restart-container.sh`](/img/restart-container.sh) script linked below to bring them up:
```
Rewrite removing pronouns

> If the containers are failing to restart, download and run the [`restart-container.sh`](/img/restart-container.sh) script to bring them up.

danciaclara commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+
+If your containers are failing to restart, you can [download and run the `restart-container.sh`](/img/restart-container.sh) script linked below to bring them up:
+
+Copy the script (`restart-container.sh`) to your installation folder and make it executable:
```
replace 'your' with 'the'

danciaclara reviewed (changes requested):
Sorry for so many comments. Most are minor nit picks to remove pronouns cc: @jnikhila

spark-c commented on website/docs/getting-started/setup/installation-guides/docker/README.mdx:
```diff
@@ -0,0 +1,199 @@
+## Prerequisites
+
+* [Docker](https://docs.docker.com/get-docker/) (version 20.10.7 or later)
+* [Docker-Compose](https://docs.docker.com/compose/install/) (version 1.29.2 or later) _(Optional)_
```
I can go ahead and remove it, although `docker-compose` isn't required. Though in reality, I'm pretty sure Docker Compose just always comes with Docker installations anyway, so anyone who has Docker should probably have Compose.

spark-c commented:
> Line 73:  If this is your first time setting up using Docker, you should expect a welcome page similar to the one below.

> I think this welcome page info is unnecessary, especially with the screenshot and all (this could change in the future). If we are including this, let's remove the pronouns
> 
> > If this is the first time that the instance has been set up using Docker, a welcome page will be displayed.
> 
> @jnikhila @sharat87 what are your thoughts?


Moved this discussion here so I could resolve that bit/get the PR ready for merge and not lose the conversation.

For the time being: I removed pronouns from that message, and then removed it in its own commit, so we can just revert that commit if we want it back.

danciaclara reviewed (dismissed):
All looks good from my end. Just want to ensure all these steps have been tested and working on both Community and Business editions. cc: @jnikhila

jnikhila commented:
> All looks good from my end. Just want to ensure all these steps have been tested and working on both Community and Business editions. cc: @jnikhila

@spark-c - As discussed earlier, could you please do a quick video recording for this?

spark-c commented:
### Test videos:
- [Setup with Docker Compose (Community Edition)](https://www.loom.com/share/5b10db14dab343ad957bf74792aad4de)
- [Update with Docker Compose (Community Edition)](https://www.loom.com/share/df32ec36b9d940c38d9754545a4caf0f)

---

- [Setup with Docker Compose (Business Edition)](https://www.loom.com/share/6ca06c47880b4c24bdf425211a9a5f74)
- ~~Update with Docker Compose (Business Edition)~~ **(identical to Community steps)**

---

- [Setup and update with Docker Run (Community Edition)](https://www.loom.com/share/8943784523664849a65f1e531e013fdb)
- [Setup and update with Docker Run (Business Edition)](https://www.loom.com/share/a26a4ab2310549999ec916c4da57932b)
